csunplugged/utils/TranslatableModelLoader.py

```diff
@@ -0,0 +1,166 @@
+"""Module for TranslatableModelLoader abstract base class."""
+
+from utils.BaseLoader import BaseLoader
+from utils.language_utils import get_available_languages, get_default_language
+from utils.errors.CouldNotFindYAMLFileError import CouldNotFindYAMLFileError
+from utils.errors.CouldNotFindMarkdownFileError import CouldNotFindMarkdownFileError
+from utils.errors.InvalidConfigValueError import InvalidConfigValueError
+from utils.errors.MissingRequiredModelsError import MissingRequiredModelsError
+from utils.errors.MissingRequiredFieldError import MissingRequiredFieldError
+from django.utils import translation
+from modeltranslation.utils import fallbacks
+
+
+class TranslatableModelLoader(BaseLoader):
+    """Abstract base class for loaders of translatable models."""
+
+    def get_yaml_translations(self, filename, field_map=None, required_slugs=[], required_fields=[]):
+        """Get a dictionary of translations for the given filename.
+
+        Yaml files must be structured as follows:
+            <object-slug>:
+                <field>: <translated value for field 2>
+                <field2>: <translated value for field 2>
+            <object-slug-2>
+                ...
```

Comment:
I think this note about the required yaml structure should be removed, this is the kind of thing that should be in the docs

Reply:
The required yaml structure will definitely be included in the docs, but I think it's also important here. I see them as serving different purposes.

- The documentation will include it for reference to anyone wishing to create new yaml files (i.e. authors)
- The function docstring outlines the type of YAML file that *this* function handles. If the file structure changes, this function will need to be updated to support that new structure (at which point the docstring would be changed to represent the new functionality).

Thoughts?

Reply:
@jordangriffiths01 Thanks for points out the differences between the two, however we don't do this for all the standard loaders (for example: this is what a lesson YAML looks like, so should we just do it in the documentation for consistency. The docstring could reference the docs also. Thoughts?

Reply:
Just chatted to @jordangriffiths01 about this. I took a lot of convincing but I'm fine for this to stay in the docstring.

My initial complaint was this was inconsistent with the standard loaders, however this method deals with YAML files in a very generic sense (there can be any number of objects, with their specific fields (of varying numbers). Therefore the documentation can detail the specifics of how to create these files for authors.

For developers, this is a crucial, but still confusing method (I still get confused by it even after reading the docstring) so having the detailed docstring plus the test cases is probably the best way to provide a clear understand of the method. Better to have extra, than less.

But the main point is that these files will also be detailed in the docs, while this snippet is to aid developers.